src/FlashLoanVault.sol

```diff
@@ -0,0 +1,285 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "lib/yield-utils-v2/contracts/mocks/ERC20Mock.sol";
+import "lib/yield-utils-v2/contracts/token/IERC20.sol";
+import "src/IERC3156FlashLender.sol";
+import "src/IERC3156FlashBorrower.sol";
+import {TransferHelper} from "lib/yield-utils-v2/contracts/token/TransferHelper.sol";
+
+/**
+@title Flash Loan Vault
+@author Calnix
+@dev Contract allows users to exchange a pre-specified ERC20 token for some other wrapped ERC20 tokens.
+@notice Wrapped tokens will be burned, when user withdraws their deposited tokens.
+*/
+
+contract FlashLoanVault is ERC20Mock, IERC3156FlashLender {
+
+    ///@dev Attach library for safetransfer methods
+    using TransferHelper for IERC20;
+
+    ///@dev The keccak256 hash of "ERC3156FlashBorrower.onFlashLoan"
+    bytes32 public constant CALLBACK_SUCCESS = keccak256("ERC3156FlashBorrower.onFlashLoan");
+
+    ///@notice Fee is a constant 0.1%
+    ///@dev 1000 == 0.1%  (1 == 0.0001 %)
+    uint256 public constant fee = 1000; 
+
+    ///@dev mapping of supported addresses by Flash loan provider
+    mapping(address => bool) public supportedTokens;
```

Comment:
You are only going to flash lend underlying, so you don't need this mapping.

Where you have to check that the flash loan is for the right token, you can just do:
`require(token == address(underlying), "FlashLender: Unsupported currency");`

Reply:
Ah okay, I was wondering about this as well. I decided to dogmatically follow the reference in the EIP since I was unsure where I could deviate, and still be considered in compliance. 

Will update this :)